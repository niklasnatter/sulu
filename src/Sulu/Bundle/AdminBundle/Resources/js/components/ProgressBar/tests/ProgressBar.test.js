// @flow
import {render, screen} from '@testing-library/react';
import React from 'react';
import ProgressBar from '../ProgressBar';

test('The component should render', () => {
    const {container} = render(
        <ProgressBar
            max={10}
            value={7}
        />
    );

    expect(container).toMatchSnapshot();
});

test('The component should render with progress skin as default', () => {
    render(
        <ProgressBar
            max={10}
            value={5}
        />
    );

    const progress = screen.queryByRole('progressbar');
    expect(progress).toHaveAttribute('max', '10');
    expect(progress).toHaveValue(5);
    expect(progress).toHaveClass('progress');
    expect(progress).toHaveTextContent('50%');
});

test('The component should render with success skin', () => {
    render(
        <ProgressBar
            max={10}
            skin="success"
            value={10}
        />
    );

    const progress = screen.queryByRole('progressbar');
    expect(progress).toHaveAttribute('max', '10');
    expect(progress).toHaveValue(10);
    expect(progress).toHaveClass('success');
    expect(progress).toHaveTextContent('100%');
});

test('The component should render with warning skin', () => {
    render(
        <ProgressBar
            max={10}
            skin="warning"
            value={0}
        />
    );

    const progress = screen.queryByRole('progressbar');
    expect(progress).toHaveAttribute('max', '10');
    expect(progress).toHaveValue(0);
    expect(progress).toHaveClass('warning');
    expect(progress).toHaveTextContent('0%');
});

test('The component should render with error skin', () => {
    render(
        <ProgressBar
            max={10}
            skin="error"
            value={3}
        />
    );

    const progress = screen.queryByRole('progressbar');
    expect(progress).toHaveAttribute('max', '10');
    expect(progress).toHaveValue(3);
    expect(progress).toHaveClass('error');
    expect(progress).toHaveTextContent('30%');
});

test('The component should render with max 0', () => {
    render(
        <ProgressBar
            max={0}
            skin="error"
            value={0}
        />
    );

    const progress = screen.queryByRole('progressbar');
    expect(progress).toHaveAttribute('max', '1');
    expect(progress).toHaveValue(0);
    expect(progress).toHaveClass('error');
    expect(progress).toHaveTextContent('0%');
});
