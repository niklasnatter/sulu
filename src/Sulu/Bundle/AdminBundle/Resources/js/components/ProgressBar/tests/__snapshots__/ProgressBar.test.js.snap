// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component should render 1`] = `
<div>
  <progress
    class="progressBar progress"
    max="10"
    value="7"
  >
    70
    %
  </progress>
</div>
`;
